tests/test_animate.py

```diff
@@ -0,0 +1,94 @@
+import time
+
+import pipx.animate
+from pipx.animate import (
+    HIDE_CURSOR,
+    CLEAR_LINE,
+    EMOJI_ANIMATION_FRAMES,
+    NONEMOJI_ANIMATION_FRAMES,
+    EMOJI_FRAME_PERIOD,
+    NONEMOJI_FRAME_PERIOD,
+)
+
+
+def check_animate_output(
+    capsys, test_string, frame_strings, frame_period, frames_to_test
+):
+    expected_string = f"{HIDE_CURSOR}" + "".join(frame_strings)
+
+    chars_to_test = 1 + len("".join(frame_strings[:frames_to_test]))
+
+    with pipx.animate.animate(test_string, do_animation=True):
+        time.sleep(frame_period * (frames_to_test - 1) + 0.1)
+    captured = capsys.readouterr()
+
+    # print for debug help if fail
+    print("expected_string:")
```

Comment:
adding '-rvva --showlocals`` achieves the same, not?

Reply:
Thanks, didn't know about that.